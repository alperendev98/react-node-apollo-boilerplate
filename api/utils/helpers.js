function getValidationRules(allowGraphqlIntrospection, graphqlDisableIntrospection) {
	if (!allowGraphqlIntrospection && !graphqlDisableIntrospection) return [];

	return allowGraphqlIntrospection ? [] : [graphqlDisableIntrospection];
}

module.exports = {
	getValidationRules
};
