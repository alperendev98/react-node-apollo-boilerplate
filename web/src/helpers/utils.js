import pickBy from 'lodash/pickBy';
import isObject from 'lodash/isObject';
import isArray from 'lodash/isArray';

export function processModifiers(modifiers) {
	return isObject(modifiers) && !isArray(modifiers) ? pickBy(modifiers) : modifiers;
}
