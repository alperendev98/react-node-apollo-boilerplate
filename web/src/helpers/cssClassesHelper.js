import BEMHelper from 'react-bem-helper';
import { processModifiers } from './utils';

export const getBEMClasses = (...classes) => {
	classes = Array.prototype.concat(...classes).filter(item => item);

	const BEMClasses = classes.map(
		className =>
			new BEMHelper({
				name: className,
				outputIsString: true
			})
	);
	return (element, modifier) =>
		BEMClasses.map(BEMClass => BEMClass(element, processModifiers(modifier))).join(' ');
};

export const addModifier = (modifiers, newModifier) => {
	let result;

	if (Array.isArray(modifiers)) {
		result = [newModifier, ...modifiers];
	} else if (typeof modifiers === 'string') {
		result = [newModifier, modifiers];
	} else if (typeof modifiers === 'object') {
		result = modifiers;
		result[newModifier] = true;
	}

	return result;
};
