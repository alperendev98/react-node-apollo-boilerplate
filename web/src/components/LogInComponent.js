import React from 'react';
import { getBEMClasses } from '../helpers/cssClassesHelper';

import '../styles/base/base.css';
import '../styles/components/loginComponent.css';

const BaseClasses = getBEMClasses(['base']);
const LoginClasses = getBEMClasses(['login']);

class LogInComponent extends React.Component {
	componentDidMount() {
		this.props.testApiConnection();
	}

	doLogin = () => {
		const { history } = this.props;

		history.push('/admin');
	};

	render() {
		return (
			<div className={LoginClasses('wrapper')}>
				<div className={LoginClasses('container')}>
					<div className={LoginClasses('title')}>Log In</div>
					<div className={LoginClasses('content')}>
						<div className={LoginClasses('btn-container')}>
							<span
								onClick={this.doLogin}
								className={BaseClasses('btn') + ' ' + BaseClasses('btn--black')}
							>
								Log In
							</span>
						</div>
					</div>
				</div>
			</div>
		);
	}
}

export default LogInComponent;
