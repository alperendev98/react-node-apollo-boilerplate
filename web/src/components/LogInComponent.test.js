import React from 'react';
import { shallow, mount } from 'enzyme';
import { getBEMClasses } from '../helpers/cssClassesHelper';

import LogInComponent from './LogInComponent';

describe('LogInComponent.js TESTS', () => {
	const props = { history: { push: jest.fn() }, testApiConnection: jest.fn() };
	let wrapper, component;

	beforeEach(() => {
		wrapper = shallow(<LogInComponent {...props} />);
		component = mount(<LogInComponent {...props} />, {});
	});

	it('renders LogInComponent without crashing', () => {
		expect(component.length).toEqual(1);
	});

	it('renders LogInComponent with .login__container class', () => {
		expect(component.find('.login__container').length).toBe(1);
	});

	it('checks the onClick when LogIn is clicked', () => {
		const instance = component.instance();
		jest.spyOn(instance, 'doLogin');

		instance.doLogin();
		expect(instance.doLogin).toHaveBeenCalled();
	});

	it('should match previos snapshot for LogInComponent', () => {
		expect(wrapper).toMatchSnapshot();
	});
});
