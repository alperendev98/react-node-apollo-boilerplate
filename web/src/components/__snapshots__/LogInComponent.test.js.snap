// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogInComponent.js TESTS should match previos snapshot for LogInComponent 1`] = `
<div
  className="login__wrapper"
>
  <div
    className="login__container"
  >
    <div
      className="login__title"
    >
      Log In
    </div>
    <div
      className="login__content"
    >
      <div
        className="login__btn-container"
      >
        <span
          className="base__btn base__btn--black"
          onClick={[Function]}
        >
          Log In
        </span>
      </div>
    </div>
  </div>
</div>
`;
